File: src/states/menu_state.py
import sys
import pygame
from src import config
from src.engine.state_machine import State
from src.engine.ui import Button

class MenuState(State):
    def __init__(self, state_machine, run_manager):
        super().__init__(state_machine, run_manager)
        self.buttons = []
        self.title_font = config.get_font("typewriter", 64)
        self.subtitle_font = config.get_font("sans", 24)
        self.desc_font = config.get_font("sans", 18)
        
        self.blinking_timer = 0
        self.show_cursor = True
        
        # Setup buttons
        start_btn = Button(
            x=config.SCREEN_WIDTH // 2 - 120,
            y=400,
            width=240,
            height=50,
            text="Start Assignment",
            callback=self.start_game,
            color=(52, 152, 219) # Vibrant blue
        )
        exit_btn = Button(
            x=config.SCREEN_WIDTH // 2 - 120,
            y=480,
            width=240,
            height=50,
            text="Resign (Exit)",
            callback=self.exit_game,
            color=(231, 76, 60) # Vibrant red
        )
        self.buttons = [start_btn, exit_btn]

    def enter(self, **kwargs):
        pass

    def start_game(self):
        self.state_machine.change_state("blind_select")

    def exit_game(self):
        pygame.quit()
        sys.exit()

    def handle_events(self, events):
        # We need to map mouse coords
        mpos = pygame.mouse.get_pos()
        # Note: If letterboxing is active, coordinate mapping is handled in main.py, 
        # so self.handle_events receives mapped coordinates.
        
        for btn in self.buttons:
            btn.check_hover(mpos)
            
        for event in events:
            for btn in self.buttons:
                if btn.handle_event(event, mpos):
                    break

    def update(self, dt):
        self.blinking_timer += dt
        if self.blinking_timer >= 0.5:
            self.show_cursor = not self.show_cursor
            self.blinking_timer = 0.0
            
        for btn in self.buttons:
            btn.update(dt)

    def draw(self, surface):
        # Fill background (Desk)
        surface.fill(config.COLOR_DESK)
        
        # Draw elegant desk grid or paper texture
        # Draw paper in center
        paper_rect = pygame.Rect(config.SCREEN_WIDTH // 2 - 350, 50, 700, 620)
        pygame.draw.rect(surface, (20, 20, 25), paper_rect.inflate(8, 8), border_radius=8)
        pygame.draw.rect(surface, config.COLOR_PAPER, paper_rect, border_radius=6)
        
        # Draw header text on paper
        title_str = "The Editor's Desk"
        if self.show_cursor:
            title_str += "_"
        title_surf = self.title_font.render(title_str, True, config.COLOR_TEXT_DARK)
        title_rect = title_surf.get_rect(center=(config.SCREEN_WIDTH // 2, 160))
        surface.blit(title_surf, title_rect)
        
        subtitle_surf = self.subtitle_font.render("A Roguelike Word-Building Adventure", True, config.COLOR_TEXT_MUTED)
        subtitle_rect = subtitle_surf.get_rect(center=(config.SCREEN_WIDTH // 2, 230))
        surface.blit(subtitle_surf, subtitle_rect)
        
        # Description text
        desc_lines = [
            "Your desk is cluttered with manuscripts.",
            "Choose your words carefully to beat target Hype scores.",
            "Draft without scoring to find letter placements.",
            "Purchase Style Guides, passive Tropes, and keyboard enhancements",
            "to survive 8 Chapters and edit the most chaotic Authors."
        ]
        
        y_off = 280
        for line in desc_lines:
            line_surf = self.desc_font.render(line, True, config.COLOR_TEXT_DARK)
            line_rect = line_surf.get_rect(center=(config.SCREEN_WIDTH // 2, y_off))
            surface.blit(line_surf, line_rect)
            y_off += 22

        # Draw buttons (which render relative to the screen coordinates)
        for btn in self.buttons:
            btn.draw(surface)

File: love2d/src/states/menu_state.lua
local config = require("src/config")
local ui = require("src/engine/ui")
local State = require("src/engine/state_machine").State

local MenuState = setmetatable({}, State)
MenuState.__index = MenuState

function MenuState.new(state_machine, run_manager)
    local self = State.new(state_machine, run_manager)
    setmetatable(self, MenuState)
    
    self.buttons = {}
    self.title_font = config.get_font("typewriter", 64)
    self.subtitle_font = config.get_font("sans", 24)
    self.desc_font = config.get_font("sans", 18)
    
    self.blinking_timer = 0
    self.show_cursor = true
    
    local start_btn = ui.Button.new(
        config.SCREEN_WIDTH / 2 - 120,
        400,
        240,
        50,
        "Start Assignment",
        function() self:start_game() end,
        {52/255, 152/255, 219/255}
    )
    local exit_btn = ui.Button.new(
        config.SCREEN_WIDTH / 2 - 120,
        480,
        240,
        50,
        "Resign (Exit)",
        function() self:exit_game() end,
        {231/255, 76/255, 60/255}
    )
    table.insert(self.buttons, start_btn)
    table.insert(self.buttons, exit_btn)
    
    return self
end

function MenuState:enter(kwargs) end

function MenuState:start_game()
    self.state_machine:change_state("assignment_select")
end

function MenuState:exit_game()
    love.event.quit()
end

function MenuState:update(dt)
    self.blinking_timer = self.blinking_timer + dt
    if self.blinking_timer >= 0.5 then
        self.show_cursor = not self.show_cursor
        self.blinking_timer = 0.0
    end
    
    local mx, my = love.mouse.getPosition()
    mx, my = config.mx or mx, config.my or my
    
    for _, btn in ipairs(self.buttons) do
        btn:check_hover(mx, my)
        btn:update(dt)
    end
end

function MenuState:mousepressed(x, y, button, istouch, presses)
    local mx, my = config.mx or x, config.my or y
    for _, btn in ipairs(self.buttons) do
        if btn:handle_event("mousepressed", mx, my, button) then
            break
        end
    end
end

function MenuState:draw()
    love.graphics.setColor(config.COLOR_DESK[1], config.COLOR_DESK[2], config.COLOR_DESK[3], 1.0)
    love.graphics.rectangle("fill", 0, 0, config.SCREEN_WIDTH, config.SCREEN_HEIGHT)
    
    local px, py, pw, ph = config.SCREEN_WIDTH / 2 - 350, 50, 700, 620
    love.graphics.setColor(20/255, 20/255, 25/255, 1.0)
    love.graphics.rectangle("fill", px - 4, py - 4, pw + 8, ph + 8, 8, 8)
    
    love.graphics.setColor(config.COLOR_PAPER[1], config.COLOR_PAPER[2], config.COLOR_PAPER[3], 1.0)
    love.graphics.rectangle("fill", px, py, pw, ph, 6, 6)
    
    local title_str = "The Editor's Desk"
    if self.show_cursor then
        title_str = title_str .. "_"
    end
    love.graphics.setFont(self.title_font)
    love.graphics.setColor(config.COLOR_TEXT_DARK[1], config.COLOR_TEXT_DARK[2], config.COLOR_TEXT_DARK[3], 1.0)
    local tw = self.title_font:getWidth(title_str)
    love.graphics.print(title_str, config.SCREEN_WIDTH / 2 - tw / 2, 130)
    
    love.graphics.setFont(self.subtitle_font)
    love.graphics.setColor(config.COLOR_TEXT_MUTED[1], config.COLOR_TEXT_MUTED[2], config.COLOR_TEXT_MUTED[3], 1.0)
    local sub_str = "A Roguelike Word-Building Adventure"
    local sw = self.subtitle_font:getWidth(sub_str)
    love.graphics.print(sub_str, config.SCREEN_WIDTH / 2 - sw / 2, 220)
    
    love.graphics.setFont(self.desc_font)
    love.graphics.setColor(config.COLOR_TEXT_DARK[1], config.COLOR_TEXT_DARK[2], config.COLOR_TEXT_DARK[3], 1.0)
    local desc_lines = {
        "Your desk is cluttered with manuscripts.",
        "Choose your words carefully to beat target Hype scores.",
        "Draft without scoring to find letter placements.",
        "Purchase Style Guides, Stationery items, and keyboard enhancements",
        "to survive 8 Chapters and edit the most chaotic Authors."
    }
    local y_off = 270
    for _, line in ipairs(desc_lines) do
        local lw = self.desc_font:getWidth(line)
        love.graphics.print(line, config.SCREEN_WIDTH / 2 - lw / 2, y_off)
        y_off = y_off + 22
    end
    
    for _, btn in ipairs(self.buttons) do
        btn:draw()
    end
end

return MenuState
